### DISCUSSION TITLE
How to add an include path to a vscode project?

### DISCUSSION THREAD

**Discussion by datascribbler777** [microsoft/vscode-discussions#198] [Extension Development QnA] [answered]
Asking this here because I can't seem to find an answer anywhere else. Really trying to give this IDE a shot, but this is a pretty basic example and it is going nowhere.

I need to see a working example of how to add an include path to vscode.

My directory structure:
```
C:.
├───.vscode
│       c_cpp_properties.json
│       launch.json
│       tasks.json
│
├───inc
│       funcs.h
│
└───src
        funcs.cpp
        main.cpp
```

funcs.h:
```
#pragma once

void sayHello();
```

funcs.cpp:
```
#include <iostream>
#include "funcs.h"

void sayHello()
{
    std::cout << "Hello!" << std::endl;
}
```

main.cpp:
```
#include "funcs.h"

int main()
{
    sayHello();
    return 0;
}
```

VSCode marks red squigglies under the #include "funcs.h" in main.cpp, but not in funcs.cpp for some reason. VSCode says it cannot find funcs.h. If I try to build the compiler throws the same error.

I have tried adding an include path for the inc folder by editing the C/C++ Configuration page and in the c_cpp_properties.json file. By default it already contains ${workspaceFolder}/** which is supposed to search recursively through all the directories. All of the googling I have tried says to add the path here but never gives the exact syntax.

I have tried adding:
/inc
./inc
../inc
/inc/*.h
${workspaceFolder}/inc
${workspaceFolder}/inc/*.h
${workspaceFolder}/inc/funcs.h
and more combinations that I would care to list here. 

How do I make vscode see my .h file in my inc folder?

**Comment by digitarald** [comment, marked as answer]
The right place to discuss CPP issues in VS Code would be the cpp extension: https://github.com/microsoft/vscode-cpptools
